### PR TITLE
T19569

### PR DESCRIPTION
**Pull request by joaquimrocha**
https://phabricator.endlessm.com/T19569

### CODE REVIEW

File: src/gs-update-monitor.c [hunk 1]
```diff
@@ -788,17 +812,36 @@ gs_update_monitor_init (GsUpdateMonitor *monitor)
...
 static void
 gs_update_monitor_dispose (GObject *object)
 {
 	GsUpdateMonitor *monitor = GS_UPDATE_MONITOR (object);
 
+	if (monitor->network_changed_handler != 0) {
+		g_signal_handler_disconnect (monitor->network_monitor,
+					     monitor->network_changed_handler);
+		monitor->network_changed_handler = 0;
+	}
```
Comment (GeorgesStavracas) [inline src/gs-update-monitor.c]: I think you should also drop the `GNetworkMonitor` reference that was taken in `init()`.

File: src/gs-update-monitor.c [hunk 2]
```diff
@@ -788,17 +812,36 @@ gs_update_monitor_init (GsUpdateMonitor *monitor)
...
+
 	if (monitor->cancellable) {
 		g_cancellable_cancel (monitor->cancellable);
 		g_clear_object (&monitor->cancellable);
 	}
+	if (monitor->network_changed_handler) {
```
Comment (GeorgesStavracas) [inline src/gs-update-monitor.c]: I guess it should be `monitor->network_cancellable` here?

File: src/gs-application.c [hunk 3]
```diff
@@ -894,6 +894,7 @@ static void
 gs_application_startup (GApplication *application)
 {
 	GSettings *settings;
+	GAction *action;
```
[the remark this answers is not in the document]
Comment (joaquimrocha) [reply src/gs-application.c]: Good catch!

File: src/gs-update-monitor.c (hunk 2 above)
Comment (joaquimrocha) [reply src/gs-update-monitor.c]: Thanks. Failed to spot the mistake in my last check.

File: src/gs-update-monitor.c (hunk 1 above)
Comment (GeorgesStavracas) [reply src/gs-update-monitor.c]: Actually, my comment is plain wrong

**Pull request merged**